app/Http/Controllers/CustomerController.php

```diff
@@ -163,6 +239,33 @@ public function export(Request $request)
         return $pdf->download('customersCustomer_'.date('Y_m_d').'.pdf');
     }
 
+    // print
+    public function print($id)
+    {
+        $accounting_system_id = $this->request->session()->get('accounting_system_id');
+        $customers = Customers::where('accounting_system_id', $accounting_system_id)
+        ->where('id', $id)
+        ->whereHas('receiptReference', function($query) {
+            $query->where('type', 'receipt')
+            ->where(function ($query) {
+                $query->where('status', 'unpaid')
+                  ->orWhere('status', 'partially_paid');
+              });})->first();
+
+        if(!$customers)
+            return redirect()->back()->with('danger', "No pending statements found.");
+
+        $receipt_reference = ReceiptReferences::where('customer_id', $id)
+        ->where('type', 'receipt')
+        ->where(function ($query) {
+            $query->where('status', 'unpaid')
+              ->orWhere('status', 'partially_paid');
+          })->get(); 
+        
+        $pdf = \PDF::loadView('customer.customer.print', compact('receipt_reference'));
+        return $pdf->download('customer_statement_'.date('Y_m_d').'.pdf');
```

Comment:
I think it is possible to open the file directly instead of requesting download.

The same goes to the other printing functions.